galley/docker/Dockerfile.galley

```diff
@@ -4,13 +4,21 @@ ARG BASE_DISTRIBUTION=default
 # The following section is used as base image if BASE_DISTRIBUTION=default
 # hadolint ignore=DL3006
 FROM istionightly/base_debug as default
+COPY galley /usr/local/bin/
+
+# 1001 is chosen as the UID to be added to the istio group as this is the UID that galley in the docker container runs as.
+RUN addgroup --system --gid 1001 istio && \
+    adduser --system --uid 1001 --gid 1001 istio && \
+    chown -R 760 /usr/local/bin/galley
+USER istio
 
 # The following section is used as base image if BASE_DISTRIBUTION=distroless
 # hadolint ignore=DL3007
-FROM gcr.io/distroless/static:latest as distroless
+FROM gcr.io/distroless/static:nonroot as distroless
+COPY galley /usr/local/bin/
```

Comment:
Seems like distroless runs as root but not-distroless doesn't? Can we make them both non-root?

Reply:
nevermind I see now we use the `:nonroot`. So what user will it run as?

Reply:
user `nonroot` was embedded in `:nonroot` image

Reply:
I have tried to set for the pod.
```
       securityContext:
            runAsUser: xxxx
```
it is working. in another word, the `istionightly/base_debug` and `:nonroot` image support changing the UID/GID at runtime